static/app/views/settings/organizationDeveloperSettings/index.tsx

```diff
@@ -59,100 +90,90 @@ class OrganizationDeveloperSettings extends AsyncView<Props, State> {
+          ) : (
+            <EmptyMessage>
+              {t('No internal integrations have been created yet.')}
+            </EmptyMessage>
+          )}
+        </PanelBody>
+      </Panel>
     );
+  }
 
-    const action = (
-      <Access organization={organization} access={['org:write']}>
-        {({hasAccess}) => (
-          <Button
-            priority="primary"
-            disabled={!hasAccess}
-            title={!hasAccess ? permissionTooltipText : undefined}
-            size="xsmall"
-            to={`/settings/${orgId}/developer-settings/new-internal/`}
-            icon={<IconAdd size="xs" isCircled />}
-          >
-            {t('New Internal Integration')}
-          </Button>
-        )}
-      </Access>
-    );
+  renderExternalIntegrations() {
+    const integrations = this.state.applications.filter(app => app.status !== 'internal');
+    const isEmpty = integrations.length === 0;
 
     return (
       <Panel>
-        <PanelHeader hasButtons>
-          {t('Internal Integrations')}
-          {action}
-        </PanelHeader>
+        <PanelHeader>{t('Public Integrations')}</PanelHeader>
         <PanelBody>
           {!isEmpty ? (
             integrations.map(this.renderApplicationRow)
           ) : (
             <EmptyMessage>
-              {t('No internal integrations have been created yet.')}
+              {t('No public integrations have been created yet.')}
             </EmptyMessage>
           )}
         </PanelBody>
       </Panel>
     );
   }
 
-  renderExternalIntegrations() {
+  renderTabContent(tab: Tab) {
+    switch (tab) {
+      case 'internal':
+        return this.renderInternalIntegrations();
+      case 'public':
+        return this.renderExternalIntegrations();
+      default:
+        return this.renderExternalIntegrations();
+    }
+  }
+  renderBody() {
     const {orgId} = this.props.params;
     const {organization} = this.props;
-    const integrations = this.state.applications.filter(app => app.status !== 'internal');
-    const isEmpty = integrations.length === 0;
 
     const permissionTooltipText = t(
-      'Manager or Owner permissions required to add a public integration.'
+      'Manager or Owner permissions required to create a new integration'
     );
 
+    const tabs = [
+      ['public', t('Public Integration')],
+      ['internal', t('Internal Integration')],
+    ] as [id: Tab, label: string][];
+
     const action = (
       <Access organization={organization} access={['org:write']}>
         {({hasAccess}) => (
           <Button
             priority="primary"
             disabled={!hasAccess}
             title={!hasAccess ? permissionTooltipText : undefined}
-            size="xsmall"
-            to={`/settings/${orgId}/developer-settings/new-public/`}
-            icon={<IconAdd size="xs" isCircled />}
+            size="small"
+            onClick={() =>
+              openCreateNewIntegrationModal({
+                orgId,
+              })
```

Comment:
nit: Can probably make the auto-formatter turn this to one line if you delete the `,` and new line after `orgId`